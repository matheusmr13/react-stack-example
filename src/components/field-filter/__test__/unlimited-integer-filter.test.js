import React from 'react';
import UnlimitedIntegerFilter from 'components/field-filter/unlimited-integer-filter';
import { shallow } from 'enzyme';

const spec = {
	id: 'myId',
	name: 'Unimited integer field'
};
describe('UnlimitedIntegerFilter component', () => {
	it('should render initial state properly', () => {
		const unlimitedIntegerFilter = shallow(<UnlimitedIntegerFilter
			spec={spec}
			onChange={() => {}}
		/>);
		expect(unlimitedIntegerFilter.find('button[disabled]')).toHaveLength(1);
		expect(unlimitedIntegerFilter).toMatchSnapshot();
	});
	it('should render properly after up click', () => {
		const onChange = jest.fn();
		const unlimitedIntegerFilter = shallow(<UnlimitedIntegerFilter
			spec={spec}
			onChange={onChange}
		/>);

		unlimitedIntegerFilter.find('button').last().simulate('click');
		expect(onChange).toHaveBeenCalled();
		expect(unlimitedIntegerFilter.find('button[disabled=false]')).toHaveLength(1);
		expect(unlimitedIntegerFilter.find('button[disabled]')).toHaveLength(1);
	});
	it('should render properly after up and down click', () => {
		const onChange = jest.fn();
		const unlimitedIntegerFilter = shallow(<UnlimitedIntegerFilter
			spec={spec}
			onChange={onChange}
		/>);

		unlimitedIntegerFilter.find('button').last().simulate('click');
		unlimitedIntegerFilter.find('button').first().simulate('click');
		expect(onChange).toHaveBeenCalledTimes(2);
		expect(unlimitedIntegerFilter.find('button[disabled]')).toHaveLength(1);
	});
});
