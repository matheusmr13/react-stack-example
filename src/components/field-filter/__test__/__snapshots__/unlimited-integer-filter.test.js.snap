// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnlimitedIntegerFilter component should render initial state properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UnlimitedIntegerFilter
    onChange={[Function]}
    spec={
        Object {
            "id": "myId",
            "name": "Unimited integer field",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="ui-filter__name"
>
          Unimited integer field
</div>,
        <div
          className="ui-filter__input"
>
          <button
                    className="ui-filter__button"
                    disabled={true}
                    onClick={[Function]}
          >
                    -
          </button>
          <span
                    className="ui-filter__counter"
          />
          <button
                    className="ui-filter__button"
                    onClick={[Function]}
          >
                    +
          </button>
</div>,
      ],
      "className": "ui-filter",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Unimited integer field",
          "className": "ui-filter__name",
        },
        "ref": null,
        "rendered": "Unimited integer field",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="ui-filter__button"
              disabled={true}
              onClick={[Function]}
>
              -
</button>,
            <span
              className="ui-filter__counter"
/>,
            <button
              className="ui-filter__button"
              onClick={[Function]}
>
              +
</button>,
          ],
          "className": "ui-filter__input",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "-",
              "className": "ui-filter__button",
              "disabled": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "-",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 0,
              "className": "ui-filter__counter",
            },
            "ref": null,
            "rendered": 0,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "+",
              "className": "ui-filter__button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "+",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ui-filter__name"
>
            Unimited integer field
</div>,
          <div
            className="ui-filter__input"
>
            <button
                        className="ui-filter__button"
                        disabled={true}
                        onClick={[Function]}
            >
                        -
            </button>
            <span
                        className="ui-filter__counter"
            />
            <button
                        className="ui-filter__button"
                        onClick={[Function]}
            >
                        +
            </button>
</div>,
        ],
        "className": "ui-filter",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Unimited integer field",
            "className": "ui-filter__name",
          },
          "ref": null,
          "rendered": "Unimited integer field",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="ui-filter__button"
                disabled={true}
                onClick={[Function]}
>
                -
</button>,
              <span
                className="ui-filter__counter"
/>,
              <button
                className="ui-filter__button"
                onClick={[Function]}
>
                +
</button>,
            ],
            "className": "ui-filter__input",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "-",
                "className": "ui-filter__button",
                "disabled": true,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "-",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 0,
                "className": "ui-filter__counter",
              },
              "ref": null,
              "rendered": 0,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "+",
                "className": "ui-filter__button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "+",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
